umbra/umbra_64.c

```diff
@@ -839,7 +875,7 @@ umbra_map_arch_init(umbra_map_t *map, umbra_map_options_t *ops)
     ASSERT(map->shadow_block_size >= ALLOC_UNIT_SIZE &&
            map->app_block_size    >= ALLOC_UNIT_SIZE,
            "block size too small");
-    map->mask = segment_mask(num_seg_bits) | seg_index_mask(num_seg_bits);
+    map->mask = IF_AARCH64_ELSE(0xfffffffffffffffc, segment_mask(num_seg_bits) | seg_index_mask(num_seg_bits));
```

Comment:
- Avoid magic values in code as this hinders readability. 
- Is it possible to have segment_mask() return the mask, rather than have the IF_AARCH64_ELSE  macro here?
- Finally, add a comment on how this mask was derived.

Reply:
I have now made these changes.